### ISSUE TITLE
Some Advice

### ISSUE THREAD

**Issue by westion717** [Rugal/oral-programming#1]
Maybe you can make a input helper to check the code context. So you can input structure by structure.  So you need a plugin in IDE to help you input and analyse the environment. For example, you say 'for var list'. It will check whether there is a variable called list and make a for loop.
Eh,That is it. Or something like that called intelligent input. It depends some work about compiler tech and there are mature libs too.

**Comment by Rugal** [comment]
I know, this is a very important part of this system.
Actually this is about nature language process.  Only after analyzing the whole sentence you input and split it into several keywords could you trigger specific commands.
Currently I am still in the phase of doing basic audio recognition and text motion jobs. I have already found a useful [website](http://demo.ark.cs.cmu.edu/parse?sentence=The%20most%20important%20thing%20is%20to%20love%20you) from CMU that could do nature language process and split it into JSON.
Any so called intelligent input needs a lot of works, that we need to do even more than that.
